transport/src/main/java/io/netty/channel/socket/nio/NioSocketChannel.java

```diff
@@ -525,4 +534,87 @@ private SocketChannel jdkChannel() {
             return ((NioSocketChannel) channel).javaChannel();
         }
     }
+
+    private interface SocketChannelOptionHelper {
+
+        boolean isSupported();
+
+        boolean isSupported(ChannelOption option);
+
+        <T> boolean setOption(SocketChannel channel, ChannelOption<T> option, T value);
+
+        <T> T getOption(SocketChannel channel, ChannelOption<T> option);
+
+        ChannelOption[] getOptions(SocketChannel channel);
+
+    }
+
+    private final class DisabledOptionHelper implements SocketChannelOptionHelper {
+
+        @Override
+        public boolean isSupported() {
+            return false;
+        }
+
+        @Override
+        public boolean isSupported(ChannelOption option) {
+            return false;
+        }
+
+        @Override
+        public <T> boolean setOption(SocketChannel channel, ChannelOption<T> option, T value) {
+            throw new IllegalArgumentException("DisabledOptionHelper");
+        }
+
+        @Override
+        public <T> T getOption(SocketChannel channel, ChannelOption<T> option) {
+            throw new IllegalArgumentException("DisabledOptionHelper");
+        }
+
+        @Override
+        public ChannelOption[] getOptions(SocketChannel channel) {
+            throw new IllegalArgumentException("DisabledOptionHelper");
+        }
+    }
+
+    private final class NioChannelOptionHelper implements SocketChannelOptionHelper {
+
+        @Override
+        public boolean isSupported() {
+            return PlatformDependent.javaVersion() >= 7;
+        }
+
+        @Override
+        public boolean isSupported(ChannelOption option) {
+            if (isSupported() && option instanceof NioChannelOption) {
```

Comment:
Just use

return isSupported() && option instanceof NioChannelOption

Reply:
I have corrected what you said. :)